packages/rest-hooks/src/globalConfig.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { createContext, ReactNode } from "react";
+import { ConfigInterface } from "swr";
+
+export type fetcherFn<R> = (...args: any) => R | Promise<R>;
+
+interface Config<D = any, Fn extends fetcherFn<D> = fetcherFn<D>> {
+  fetcher?: Fn;
+}
+
+export const GlobalConfigContext = createContext<Config | undefined>(undefined);
+
+type GlobalConfigProps = {
+  config?: Config;
+  children: ReactNode;
+};
+
+// Configuration priorities from highest to lowest (currently only for fetching)
```

Comment:
On second thought this was stupid, it still doesn't solve our PUT/POST/PATCH issue :|

Maybe we should take in the BASE_URL instead